api/src/main/java/org/openmrs/Cohort.java

```diff
@@ -242,9 +249,9 @@ public static Cohort intersect(Cohort a, Cohort b) {
 	public static Cohort subtract(Cohort a, Cohort b) {
 		Cohort ret = new Cohort();
 		if (a != null) {
-			ret.getMemberIds().addAll(a.getMemberIds());
+			ret.getMembers().addAll(a.getMembers());
 			if (b != null) {
-				ret.getMemberIds().removeAll(b.getMemberIds());
+                ret.getMembers().removeAll(b.getMembers());
```

Comment:
@wluyima I just have a small doubt. In the equals() and hashCode(), should be I checking the object's uuid equality and hashCode? Something like this?

```
	@Override
	public int hashCode() {
        if (this.getUuid() == null) {
			return super.hashCode();
		} else {
			return this.getUuid().hashCode();
		}
	}

	@Override
	public boolean equals(Object obj) {
		if (obj == null) {
			return false;
		}
		if (!(obj instanceof CohortMember)) {
			return false;
		}
		CohortMember other = (CohortMember) obj;
		return Objects.equals(this.getUuid(), other.getUuid());
	}
```